examples/mxnet/gcn/gcn_spmv.py

```diff
@@ -0,0 +1,206 @@
+"""
+Semi-Supervised Classification with Graph Convolutional Networks
+Paper: https://arxiv.org/abs/1609.02907
+Code: https://github.com/tkipf/gcn
+
+GCN with SPMV optimization
+"""
+import argparse, time, math
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+import dgl
+from dgl import DGLGraph
+import dgl.function as fn
+from dgl.data import register_data_args, load_data
+
+class GCNLayer(gluon.Block):
+    def __init__(self,
+                 g,
+                 in_feats,
+                 out_feats,
+                 activation,
+                 dropout,
+                 bias=True):
+        super(GCNLayer, self).__init__()
+        self.g = g
+        with self.name_scope():
+            self.weight = self.params.get('weight', shape=(in_feats, out_feats),
+                    init=mx.init.Xavier())
+            if bias:
+                self.bias = self.params.get('bias', shape=(out_feats,),
+                    init=mx.init.Zero())
+            else:
+                self.bias = None
```

Comment:
why not use gluon Linear layer?

Reply:
Simply follows the codes here: https://github.com/tkipf/pygcn/blob/4396e4db5b97c9e071516bc601c9b6693696c489/pygcn/layers.py#L31-L37 . The weight is used before message passing while the bias is used afterwards.

Reply:
interesting. do you know any reason why it should be done like this?

Reply:
The math is `relu(A * H * W + b)`, where `A` is the normalized adj, `H` is the feature, `W` is the weight, `b` is the bias. There could be two implementations: (1) do message passing first (A * H), then apply a dense layer. (2) Do H * W first, then message passing, then add bias. I guess why the author chose the later one is because the input dimension is much bigger than the hidden dimension, so H*W first makes the computation more efficient.

Reply:
i see. thanks for the explanation.